.travis.yml

```diff
@@ -1,26 +1,58 @@
 language: cpp
+compiler: gcc
+os: linux
+dist: bionic
+cache: ccache
 
-compiler:
-  - clang
-  - gcc
+addons:
+    apt:
+        sources:
+            - ubuntu-toolchain-r-test
+        packages:
+            - g++-4.8
+            - cmake
+            - ninja-build
+            - ccache
+            - libx11-dev
+            - libxcursor-dev
 
 env:
-  matrix:
-    - SHARED=OFF
-    - SHARED=ON
+  - SHARED=OFF
 
 before_install:
-  - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
-  - sudo apt-get update -qq
   - if [ "$SHARED" == "ON" ]; then sudo apt-get install -y libcurl4-openssl-dev libgif-dev libfreetype6-dev libjpeg-dev libz-dev libpng-dev libtinyxml-dev libpixman-1-dev liballegro4.2-dev ; fi
-  - sudo apt-get install -y -qq g++-4.8
-  - if [ "$CXX" = "g++" ]; then export CXX="g++-4.8"; fi
+  - if [ "${TRAVIS_OS_NAME}" = "linux" ] && [ "$CXX" = "g++" ]; then export CXX="g++-4.8"; fi
 
 before_script:
+  - ccache -z
   - mkdir build
   - cd build
-  - cmake .. -DUSE_SHARED_CURL=$SHARED -DUSE_SHARED_GIFLIB=$SHARED -DUSE_SHARED_FREETYPE=$SHARED -DUSE_SHARED_JPEGLIB=$SHARED -DUSE_SHARED_ZLIB=$SHARED -DUSE_SHARED_LIBPNG=$SHARED -DUSE_SHARED_TINYXML=$SHARED -DUSE_SHARED_PIXMAN=$SHARED -DUSE_SHARED_ALLEGRO4=$SHARED -DENABLE_TESTS=ON
+  - cmake -G Ninja .. -DCMAKE_INSTALL_PREFIX=${TRAVIS_BUILD_DIR}/build/install -DUSE_SHARED_CURL=$SHARED -DUSE_SHARED_GIFLIB=$SHARED -DUSE_SHARED_FREETYPE=$SHARED -DUSE_SHARED_JPEGLIB=$SHARED -DUSE_SHARED_ZLIB=$SHARED -DUSE_SHARED_LIBPNG=$SHARED -DUSE_SHARED_TINYXML=$SHARED -DUSE_SHARED_PIXMAN=$SHARED -DUSE_SHARED_ALLEGRO4=$SHARED -DENABLE_TESTS=ON
 
 script:
-  - "make"
-  - "ctest --output-on-failure"
+  - ninja libresprite
+  - ninja install
+  - cd "${TRAVIS_BUILD_DIR}"
+  - ccache -s
+
+after_success:
+  - cd build
+  - wget https://github.com/linuxdeploy/linuxdeploy/releases/download/continuous/linuxdeploy-x86_64.AppImage
+  - chmod +x linuxdeploy-x86_64.AppImage
+  - cp ../.LibreSprite.desktop LibreSprite.desktop
+  - ./linuxdeploy-x86_64.AppImage --appdir AppDir -e install/bin/libresprite -i install/share/libresprite/data/icons/ase48.png -d LibreSprite.desktop
```

Comment:
I think `ninja install` would give you `install/share/applications/libresprite.desktop` ready to use. Not sure about the icon though, from a quick look on GitHub I didn't see the install instructions for the icon, but we I might have missed it. If not, we should fix that (it should install the `ase*.png` icons in `/usr/share/icons/hicolor/*/apps/libresprite.png`).